src/jetzig/http/Session.zig

```diff
@@ -91,97 +75,70 @@ fn save(self: *Self) !void {
     if (self.state != .parsed) return error.UnparsedSessionCookie;
 
     const json = try self.data.toJson();
-    defer self.allocator.free(json);
-
-    if (self.encrypted) |*ptr| {
-        self.allocator.free(ptr.*);
-        self.encrypted = null;
-    }
-    self.encrypted = try self.encrypt(json);
 
-    const encoded = try jetzig.util.base64Encode(self.allocator, self.encrypted.?);
+    const encrypted = try self.encrypt(json);
+    defer self.allocator.free(encrypted);
+    const encoded = try jetzig.util.base64Encode(self.allocator, encrypted);
     defer self.allocator.free(encoded);
-
-    if (self.cookie) |*ptr| self.allocator.free(ptr.*.value);
-    self.cookie = .{ .value = try self.allocator.dupe(u8, encoded) };
-
-    try self.cookies.put(
-        cookie_name,
-        self.cookie.?,
-    );
+    std.debug.print("encoded: {s}\n", .{encoded});
+    try self.cookies.put(cookie_name, .{ .value = encoded });
 }
 
 fn parseSessionCookie(self: *Self, cookie_value: []const u8) !void {
-    self.data = jetzig.data.Data.init(self.allocator);
     const decoded = try jetzig.util.base64Decode(self.allocator, cookie_value);
     defer self.allocator.free(decoded);
 
-    const buf = self.decrypt(decoded) catch |err| {
+    const decrypted = self.decrypt(decoded) catch |err| {
         switch (err) {
             error.AuthenticationFailed => return error.JetzigInvalidSessionCookie,
             else => return err,
         }
     };
-    defer self.allocator.free(buf);
-    if (self.decrypted) |*ptr| self.allocator.free(ptr.*);
-    self.decrypted = try self.allocator.dupe(u8, buf);
+    defer self.allocator.free(decrypted);
 
-    try self.data.fromJson(self.decrypted.?);
+    try self.data.fromJson(decrypted);
     self.state = .parsed;
 }
 
-fn decrypt(self: *Self, data: []const u8) ![]const u8 {
-    if (self.encryption_key) |secret| {
-        const encrypted = data[0 .. data.len - Cipher.tag_length];
-        const secret_bytes = std.mem.sliceAsBytes(secret);
-        const key: [Cipher.key_length]u8 = secret_bytes[0..Cipher.key_length].*;
-        const nonce: [Cipher.nonce_length]u8 = secret_bytes[Cipher.key_length .. Cipher.key_length + Cipher.nonce_length].*;
-        const buf = try self.allocator.alloc(u8, data.len - Cipher.tag_length);
-        const additional_data = "";
-        var tag: [Cipher.tag_length]u8 = undefined;
-        std.mem.copyForwards(u8, &tag, data[data.len - Cipher.tag_length ..]);
-
-        try Cipher.decrypt(
-            buf,
-            encrypted,
-            tag,
-            additional_data,
-            nonce,
-            key,
-        );
-        return buf[0..];
-    } else {
-        return self.allocator.dupe(u8, "hello");
-    }
+fn decrypt(self: *Self, data: []u8) ![]u8 {
+    const secret_bytes = std.mem.sliceAsBytes(self.encryption_key);
+    const key = secret_bytes[0..Cipher.key_length];
+    const nonce = data[0..Cipher.nonce_length];
```

Comment:
Does this panic when `data` is too short?

Reply:
Yeah, but that's annoying - I'll add some runtime bounds checks to make it more friendly.